client/src/controllers/BulkController.ts

```diff
@@ -0,0 +1,69 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Controller that adds the ability to toggle some or all the elements, ignoring the disabled ones
+ * Additionaly, it checks if all non-disabled items are selected or not. If they are, then select all
+ *
+ * @example
+ * <div data-controller="w-bulk">
+ *   <input type="checkbox" data-w-bulk-target="all">
+ *
+ *   <div>
+ *     <input type="checkbox" data-w-bulk-target="item" disabled>
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *     <input type="checkbox" data-w-bulk-target="item">
+ *   </div>
+ *
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
+ *   <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
+ * </div>
+ */
+export class BulkController extends Controller {
+  /** The list of targets for this controller. */
+  static targets = ['all', 'item'];
+
+  /** The list of all select-all checkbox targets for this controller. */
+  declare readonly allTargets: HTMLInputElement[];
+
+  /** The list of all item checkbox targets for this controller. */
+  declare readonly itemTargets: HTMLInputElement[];
+
+  get activeItems() {
+    return this.itemTargets.filter(({ disabled }) => !disabled);
+  }
+
+  /**
+   * Toggles the select-all checkbox based on item checkboxes.
+   */
+  toggle(): void {
+    const isAllChecked = !this.activeItems.some((item) => !item.checked);
+
+    // Update all select all checkboxes (except disabled ones)
+    this.allTargets.forEach((target) => {
+      if (!target.disabled) {
+        // eslint-disable-next-line no-param-reassign
+        target.checked = isAllChecked;
+      }
+    });
+  }
+
+  /**
+   * Toggles all item checkboxes based on select-all checkbox.
+   */
+  toggleAll(event): void {
+    const force = event.params.get('force');
+    const isChecked = force != null ? force : event.target.checked;
+
```

Comment:
```suggestion
  /**
   * Toggles all item checkboxes based on select-all checkbox.
   */
  toggleAll(event: Event & { params?: { force?: boolean } }): void {
    const force = event?.params?.force;
    const checkbox = event.target as HTMLInputElement;
    const isChecked = typeof force === 'boolean' ? force : checkbox.checked;

```

* Use types for the param here, we are saying it can be an event AND this event may have a `params` object
* calling `get` on the object is probably a bit risky, may not work in some cases, so let's simplify this code 
* Finally, you want to check that the type is `boolean`, checking it's not null is tricky as it could be undefined